IPython/utils/openpy.py

```diff
@@ -72,7 +73,8 @@ def read_py_file(filename, skip_encoding_cookie=True):
     -------
     A unicode string containing the contents of the file.
     """
-    with open(filename) as f:   # the open function defined in this module.
+    filepath = Path(filename)
+    with filepath.open() as f:   # the open function defined in this module.
```

Comment:
```suggestion
    with open(filepath) as f:   # the open function defined in this module.
```

Slightly tricky as `open` here is not the builtin `open`; hence why test were failing.